internal/app/service/achievement_service.go

```diff
@@ -5,13 +5,15 @@ import (
 
 	"github.com/joshsoftware/profile_builder_backend_go/internal/pkg/dto"
 	"github.com/joshsoftware/profile_builder_backend_go/internal/pkg/errors"
+	"github.com/joshsoftware/profile_builder_backend_go/internal/pkg/helpers"
 	"github.com/joshsoftware/profile_builder_backend_go/internal/repository"
 	"go.uber.org/zap"
 )
 
 // AchievementService represents a set of methods for accessing the achievements
 type AchievementService interface {
 	CreateAchievement(ctx context.Context, cDetail dto.CreateAchievementRequest) (profileID int, err error)
+	GetAchievements(ctx context.Context, profileID string) (value []dto.AchievementResponse, err error)
```

Comment:
Add ListFilter in method signature 

type ListFilter  struct {
   ActiveMentIDs []int
  Name []string
}

Reply:
kk